### DISCUSSION TITLE
How to fix a broken flow passed to Prefect Agent?

### DISCUSSION THREAD

**Discussion by Srj** [PrefectHQ/prefect#3769] [Q&A] [answered]
Suppose, I have a flow that raises some unexpected exception at some point while running and i have to fix that task function and restart from where it failed without starting from the beginning. How can i do it using Prefect UI? I am passing my flow to an Agent. So is there a way to fix it with updating the flow and start over again?

For example,
```
@task
def inc(x):
    time.sleep(1)
    return x + 1


@task
def dec(x):
    time.sleep(2)
    return x - 1


@task
def add(x, y):
    time.sleep(1)
    return x + y


@task
def list_sum(arr):
    time.sleep(1)
    raise Exception()                ###Will cause an exception
    return sum(arr)


with Flow("dask-example") as flow:
    incs = inc.map(x=range(100))
    decs = dec.map(x=range(100))
    adds = add.map(x=incs, y=decs)
    total = list_sum(adds)
```

I have already passed this to Prefect Agent. Now how can i edit this error and start from where it broke. Thanks

**Comment by joshmeek** [comment, marked as answer]
Hi @Srj look into using Prefect's file-based flow storage mechanism [here](https://docs.prefect.io/core/idioms/file-based.html). This will allow you to update your flow code directly and rerun without needing to reregister a new version of the flow.

**Reply by Srj** [reply]
Thanks a lot. That's what i have been looking for.

**Reply by vlcinsky** [reply]
The link with file-based flow store mechanism https://docs.prefect.io/core/idioms/file-based.html shows 404.

I am unable to find similar topic in existing documentation.

It would be great to have valid link, but even if an option to repair code of existing flow is not available any more, information about not being avaliable is valuable too.

**Reply by zanieb** [reply]
https://docs.prefect.io/orchestration/flow_config/storage.html#pickle-vs-script-based-storage
